crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -2324,7 +2324,7 @@ impl<'a> Checker<'a> {
                     let parsed_expr = parsed_annotation.expression();
                     self.visit_expr(parsed_expr);
                     if self.semantic.in_type_alias_value() {
-                        if self.enabled(Rule::QuotedTypeAlias) {
+                        if !self.source_type.is_stub() && self.enabled(Rule::QuotedTypeAlias) {
```

Comment:
Let's add a comment so it's obvious to future readers why this rule is disabled on stub files:

```suggestion
                        // stub files are covered by PYI020
                        if !self.source_type.is_stub() && self.enabled(Rule::QuotedTypeAlias) {
```